diff --git a/menu_token.module b/menu_token.module
index c530cc5..ecc9f96 100644
--- a/menu_token.module
+++ b/menu_token.module
@@ -181,6 +181,59 @@ function menu_token_menu_link_alter(&$item) {
 }
 
 /**
+ * Find a link path based on a router path.
+ * @param  $router_path
+ * @return The link path.
+ */
+function menu_token_find_link_path($router_path){
+  $query = db_select('menu_links', 'ml');
+  $query->fields('ml', array('link_path'))
+  ->condition('ml.router_path', 'menutoken/%')
+  ->condition('ml.options', '%' . $router_path . '%', 'LIKE');
+  $link_path = $query->execute()->fetchField();
+  return $link_path;
+}
+
+/**
+ * Implements hook_query_alter().
+ *
+ * No active trail is set when using paths with tokens, as the tokens are
+ * replaced, and the correct path is not found in the menu link table. To fix
+ * this, we replace the resolved path in the the preferred menu query's
+ * conditions with menutoken one. Also replace the path that will be returned
+ * with the resolved path, to allow it to be matched with the candidates in
+ * menu_link_get_preferred().
+ *
+ * @see menu_link_get_preferred()
+ */
+function menu_token_query_alter(QueryAlterableInterface $query) {
+  if ($query->hasTag('preferred_menu_links')) {
+    // Get all the conditions as a reference.
+    $conditions = &$query->conditions();
+
+    foreach ($conditions as &$condition) {
+      // If the current condition is the link_path one
+      if (is_array($condition) && isset($condition['field']) && $condition['field'] == 'ml.link_path') {
+        foreach ($condition['value'] as $path_key => $path_value) {
+          // If the path has a coresponding menutoken path
+          if ($menutoken_path = menu_token_find_link_path($path_value)) {
+            // Replace the condition value with the menutoken path.
+            $condition['value'][$path_key] = $menutoken_path;
+
+            // Add the resolved link path as an expression to the query. This
+            // will replace the field link_path from the table menu_links with
+            // the resolved path, allowing menu_link_get_preferred() to match it
+            // based on data available in that function.
+            $query->addExpression("'$path_value'", 'link_path');
+          }
+        }
+      }
+    }
+  }
+}
+
+
+/**
  * Returns TRUE if 'menu_links_features_export_render' is in the callstack.
  */
 function _menu_token_is_called_from_features() {
