diff --git a/sites/all/modules/logintoboggan/logintoboggan.module b/sites/all/modules/logintoboggan/logintoboggan.module
index 55b93fe..2dd63d3 100644
--- a/sites/all/modules/logintoboggan/logintoboggan.module
+++ b/sites/all/modules/logintoboggan/logintoboggan.module
@@ -523,7 +523,10 @@ function logintoboggan_user_register_submit($form, &$form_state) {
   // Mail the user.
   _user_mail_notify($mailkey, $account);
 
-  drupal_set_message($message);
+  //stop message from being show on register for agencies
+  if(!isset($account->profile_agency)) {
+    drupal_set_message($message);
+  }
 
   // where do we need to redirect after registration?
   $redirect = _logintoboggan_process_redirect(variable_get('logintoboggan_redirect_on_register', ''), $account);
